### ISSUE TITLE
edition 2018 support for mod #[path=] attribute

### ISSUE THREAD

**Issue by burrbull** [rust-lang/rust#59345]
If module files have edition 2018 style file structure and try to import one of them by path attribute:
```
#[path=modname.rs] 
mod modname;
```
compiler tries to find submodules in submodname.rs or submodname/mod.rs, but not in
**modname**/submodname.rs where it is really located.

**Comment by crlf0710** [comment]
Just run into this issue with `winit` 2018 migration.  cc #61924 https://github.com/rust-windowing/winit/issues/912

**Comment by crlf0710** [comment]
@petrochenkov do you think this is a `C-bug` or `C-enhancement`?

**Comment by petrochenkov** [comment]
>do you think this is a C-bug or C-enhancement?

¯\_(ツ)_/¯
I always forget how exactly `#[path]` works, I only remember that it doesn't work consistently, but changing that is a breaking change.

**Comment by crlf0710** [comment]
Is it a breaking change adding an extra candidate?

**Comment by petrochenkov** [comment]
cc @cramertj (perhaps you remember the details?)

**Comment by cramertj** [comment]
Yeah, it's a totally confusing mess. [The reference's section on this](https://doc.rust-lang.org/reference/items/modules.html#the-path-attribute) should be correct, though.

**Comment by ehuss** [comment]
The reference is incomplete. See https://github.com/rust-lang-nursery/reference/issues/573, which I believe is exactly this issue.
